src/main/java/zd/zero/waifu/motivator/plugin/assets/AssetManager.kt

```diff
@@ -75,11 +76,16 @@ object AssetManager {
 
     private fun resolveTheAssetUrl(localAssetPath: Path, remoteAssetUrl: String): Optional<String> =
         when {
-            hasAssetChanged(localAssetPath, remoteAssetUrl) ->
-                downloadAndGetAssetUrl(localAssetPath, remoteAssetUrl)
-            Files.exists(localAssetPath) ->
+            Files.exists(localAssetPath) -> {
+                ApplicationManager.getApplication().executeOnPooledThread {
+                    if (hasAssetChanged(localAssetPath, remoteAssetUrl)) {
+                        downloadAndGetAssetUrl(localAssetPath, remoteAssetUrl)
+                    }
+                }
+
                 localAssetPath.toUri().toString().toOptional()
-            else -> Optional.empty()
+            }
+            else -> downloadAndGetAssetUrl(localAssetPath, remoteAssetUrl)
```

Comment:
If it doesn't exist, fallback to the original behavior, to download and use the asset. This will only happen if it's the first asset for the chosen category.